proposed/2021/VisualStudioPackageInstallDeepLink.md

```diff
@@ -0,0 +1,163 @@
+3.	Solution level PMUI is already open.
+4. Go to Browse tab of the solution level PMUI.
+    1. Search query for [PackageId: “PackageName”] is prepopulated – yielding only the target package as a result.
+    2. Target package should be pre-selected with its package details open to the target version from NuGet.org.
+5. User chooses relevant projects and installs the package as desired.
+
+### Scenario 5 - Project level PMUI is open
+
+1.	Deep link URL is clicked.
+2.	Solution is already selected and open.
+3.	Project level PMUI is already open.
+4. In the PMUI window that is currently open/ in focus, open the Browse tab.
+    1. Search query for [PackageId: “PackageName”] is prepopulated – yielding only the target package as a result.
+    2. Target package should be pre-selected with its package details open to the target version from NuGet.org.
+   ![Project level PMUI Browse tab filtered to target package](./meta/resources/VisualSutdioPackageInstallDeepLink/Project%20level%20filtered%20package%20results.png)
+5. User chooses relevant projects and installs the package as desired.
+
+## Drawbacks
+
+* Older versions of Visual Studio won't support this.
+* Current proposal is limited to NuGet.org packages - generalizing to packages from any feed will increase engineering complexity significantly as we would need a custom way to isolate the target package in the Browse tab.
+
+## Prior Art
+
+### Download VS Code extension from VS Marketplace
+
+Example Extension: [Dracula Official - Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=dracula-theme.theme-dracula)
+
+Deep link for Dracula Official: [vscode:extension/dracula-theme.theme-dracula](vscode:extension/dracula-theme.theme-dracula)
+
+Behavior: Click the deep link opens VS Code and the extension details page for the target extension, making it a 1 click install experience from there.
+
+![Download Dracula Official from VS Marketplace website](./meta/resources/VisualSutdioPackageInstallDeepLink/Dracula%20Official.png)
+
+## Unresolved Questions
+
+**Q: What should the structure of the deep link URL?**
+
+**Q: If a project level PMUI window is already open or in focus, should we open the package there or open the solution level PMUI anyway for a consistent experience?**
+
+*Pending customer development*
+
+**Q: What happens if NuGet.org is not enabled as a source?**
+
+* Option 1: Open a dialogue explaining to the customer “You are attempting to install a package from NuGet.org but you do not have it enabled as a source. Would you like to enable it or cancel?”
+* Option 2: Switch feed setting to “All” automatically
+
+**Q: What package types should support downloading in Visual Studio (library, template, etc.)?**
+
+Currently, anything but library packages fail to install through the PMUI and don’t make sense to install through the PMUI. 
+
+**Q: What Visual Studio operation can we use as prior art?**
+
+•	Open repo in Visual Studio on GitHub.
```

Comment:
This was an experiment for Visual Studio Codespaces...I think this actually did have some detection of VS installed -- should check with the team there and if any header information was used, etc. That would be good to 'light up' the option knowing you have some rate of success versus offering an option where someone may not have VS (or the supported version of VS) installed and send them on a wild chase.